core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -142,6 +141,10 @@ case class FetchPartitionData(error: Errors = Errors.NONE,
                               preferredReadReplica: Option[Int],
                               isReassignmentFetch: Boolean)
 
+case class FencedPartitionMetadata(partition: Partition,
+                                   metadataPartition: MetadataPartition,
```

Comment:
nit: could we just call this field `metadata`? I think the `Partition` suffix is implied from the context.